public/app/features/logs/utils.ts

```diff
@@ -138,9 +138,34 @@ export const sortLogsResult = (logsResult: LogsModel | null, sortOrder: LogsSort
   return logsResult ? { ...logsResult, rows } : { hasUniqueLabels: false, rows };
 };
 
-export const sortLogRows = (logRows: LogRowModel[], sortOrder: LogsSortOrder) =>
-  sortOrder === LogsSortOrder.Ascending ? logRows.sort(sortInAscendingOrder) : logRows.sort(sortInDescendingOrder);
+export const sortLogRows = (logRows: LogRowModel[], sortOrder: LogsSortOrder): LogRowModel[]  => {
```

Comment:
I'm curious why you implemented this change here, but this is not the place for the fix nor a correct implementation. There was already a sorting implementation in the original function:   return sortOrder === LogsSortOrder.Ascending ? logRows.sort(sortInAscendingOrder) : logRows.sort(sortInDescendingOrder);

The place where the sorting should occur is https://github.com/grafana/grafana/blob/main/public/app/features/explore/Logs/LogsMetaRow.tsx#L77 . Please give this a try and double check that, for each format, the visible order matches the download order.

Reply:
thanks for your review. i will try fix this issue at
https://github.com/grafana/grafana/blob/main/public/app/features/explore/Logs/LogsMetaRow.tsx#L77